### ISSUE TITLE
MQTT Output Binding with Mosquitto not working 

### ISSUE THREAD

**Issue by tareksharbak** [dapr/components-contrib#850]
## Expected Behavior

Calling the output binding API with output binding configured to use MQTT with a certain topic:
- Get 204 HTTP status code
- Message is published on the topic

## Actual Behavior

Calling the output binding API with output binding configured to use MQTT with a certain topic:
- Get 204 HTTP status code
- Message is **NOT** published on the topic

## Steps to Reproduce the Problem

-  Create a new output binding yaml file:
`apiVersion: dapr.io/v1alpha1`
`kind: Component`
`metadata:`
`  name: outputmqtt`
`spec:`
`  type: bindings.mqtt`
`  version: v1`
`  metadata:`
`  - name: url`
`    value: "tcp://localhost:1883"`
`  - name: topic`
`    value: foo`
- Start dapr in self-hosted mode without an app command:
`dapr run --log-level debug --app-id testapp --components-path ./components --dapr-http-port 3550`
- Call the output binding api
`curl -X POST -H 'Content-Type: application/json' http://localhost:3550/v1.0/bindings/outputmqtt -d '{ "data": { "message": "Hi!" }, "operation": "create" }'`
- Check to see if message was published to topic: 'foo'

I also tried substituting the output binding with Redis, which worked as expected. 
I attached my dapr logs and the components files that I was using for the test (I appended .txt to the files to be able to upload them here).

[outputmqtt.yaml.txt](https://github.com/dapr/components-contrib/files/6400840/outputmqtt.yaml.txt)
[outputredis.yaml.txt](https://github.com/dapr/components-contrib/files/6400841/outputredis.yaml.txt)
[log.txt](https://github.com/dapr/components-contrib/files/6400838/log.txt)
[commands.txt](https://github.com/dapr/components-contrib/files/6400837/commands.txt)

## Release Note

RELEASE NOTE:

**Comment by pkedy** [comment]
Hi, @tareksharbak. It is normal for MQTT brokers to not store messages for delivery until there is at least 1 consumer is subscribed to them. Can you please confirm that you have a consumer subscribed to the topic before publishing the message? From your logs this may be happening, but there might be some race condition going on still. Perhaps try to have the app sleep on startup, then publish a message well after you know there is a subscriber. If you still are not seeing the message after that, then there is definitely a bug.

**Comment by tareksharbak** [comment]
Hi, this seems to have been fixed in the latest 1.2.0 release.
When starting dapr v1.2.0 with the binding yaml file I posted above, it complains that a CosumerID is missing (The docs should be updated to show that the ConsumerID is now required).

In any case, it's great to see that v1.2.0 fixed the issue with the MQTT output binding.

@pkedy I did have a consumer subscribed to the topic, it was how I was checking if the message is published or not. The issue seems to be only affecting the older version of the runtime v1.1.0, now it's fixed with the new one (v1.2.0) 😃
